### PR TITLE
Hotfix yy account test

### PR DESCRIPTION
**Pull request by YuqingYoki**
Failed tests:
//testing empty PrimaryEamilAddress and existing accountEmailAddress
//Testing empty primaryPhoneNumber but existing phoneNumber

---
This change is [Reviewable](https://reviewable.io/reviews/ten24/slatwall/4466)

### CODE REVIEW

File: meta/tests/unit/entity/AccountTest.cfc [hunk 1]
```diff
@@ -121,6 +121,131 @@ component extends="Slatwall.meta.tests.unit.entity.SlatwallEntityTestBase" {
...
+			primaryEmailAddress = {}
+		};
+		mockAccount = createTestEntity('Account', accountData);
+		assertTrue(isNull(mockAccount.getPrimaryEmailAddress().getEmailAddress()));
+		//testing empty PrimaryEamilAddress and existing accountEmailAddress
+		accountData = {
+			accountID = "001",
+			firstName = "Hello",
+			lastName = "Kitty",
+			primaryEmailAddress = {},
```
Comment (jasperboyd) [inline meta/tests/unit/entity/AccountTest.cfc]: Your primaryEmailAddress is an empty struct {} after you create the mockAccount you need to set the primaryEmailAddress to one of the account addresses you make below

**Comment by YuqingYoki** [comment]
```
if(!isNull(variables.primaryEmailAddress)) {
	return variables.primaryEmailAddress;
 } else if (arrayLen(getAccountEmailAddresses())) {
```
For the //testing empty PrimaryEamilAddress and existing accountEmailAddress case, I was try to test this ELSE IF condition. The `arrayLen(getAccountEmailAddresses() `should be 2 right?

**Comment by jasperboyd** [comment]
@YuqingYoki yep you're right about that, but because you defined primaryEmailAddress as a struct it's hitting the first if condition and returning the {} which doesn't have an email address, to test what you're describing delete the line where you set primaryEmailAddress to struct  https://github.com/ten24/slatwall/blob/master/model/entity/Account.cfc#L631

File: meta/tests/unit/entity/AccountTest.cfc [hunk 2]
```diff
@@ -121,6 +121,131 @@ component extends="Slatwall.meta.tests.unit.entity.SlatwallEntityTestBase" {
...
+					emailAddress = "firstaccountEamilAddress@hotmail.com"
+				},{
+					accountEmailAddressID = "00034",
+					emailAddress = "secondAccountEmailAddress@hotmail.com"
+				}
+			]		
+		};
+		var mockAccount = createTestEntity('Account', accountData);
+		request.debug(mockAccount);//??why AddressEmailAccount is Empty
+		var resultEmailAddress = mockAccount.getEmailAddress();
```
Comment (rmarchandTen24) [inline meta/tests/unit/entity/AccountTest.cfc]: getEmailAddress looks for primary email address

public string function getEmailAddress() {
	return getPrimaryEmailAddress().getEmailAddress();
}

your mock data should have 

primaryEmailAddress={
     accountEmailAddressid="",
     emailAddress="myemail@company.com"
}

File: meta/tests/unit/entity/AccountTest.cfc (hunk 1 above)
Comment (rmarchandTen24) [inline meta/tests/unit/entity/AccountTest.cfc]: @jasperboyd  hit this nail on the head. If it is empty then it should not even be declared. @YuqingYoki a user can have many accountEmailAddresses however that does not mean that they have a primaryEmailAddress yet. Again you must mock that data.

File: meta/tests/unit/entity/AccountTest.cfc [hunk 3]
```diff
@@ -121,6 +121,131 @@ component extends="Slatwall.meta.tests.unit.entity.SlatwallEntityTestBase" {
...
+		};
+		var mockAccount = createTestEntity('Account', accountData);
+		var resultPrimaryPhone = mockAccount.getPrimaryPhoneNumber().getPhoneNumber();
+		assertEquals(resultPrimaryPhone, "123");
+		//Testing emmpty primaryPhoneNumber
+		accountData = {
+			accountID = "001",
+			firstName = "Hello",
+			lastName = "Kitty",
+			primaryPhoneNumber = {}
```
Comment (rmarchandTen24) [inline meta/tests/unit/entity/AccountTest.cfc]: if you are testing an missing primaryPhoneNumber. Do not mock the data and then 
`assert(isNull(mockAccount.getPrimaryPhoneNumber()))`

**Pull request closed without merging**